### PR TITLE
IE fix: allow add .app.domain.com after .domain.com without SystemPropertiesManager.getAsBoolean(Constants.SET_COOKIE_TO_ALL_DOMAINS, true)          

### PR DESCRIPTION
**Pull request by vharseko**

### CODE REVIEW

File: openam-shared/src/main/java/com/sun/identity/shared/encode/CookieUtils.java [hunk 1]
```diff
@@ -474,10 +476,17 @@ public static void addCookieToResponse(HttpServletResponse response, Cookie cook
         Set<String> domains = new HashSet<>();
 
         for (String domain : cookieDomains) {
-            if (domain == null || host.contains(domain)) {
+            if (domain != null && StringUtils.endsWithIgnoreCase(host, domain)) {
```
Comment (maximthomas) [inline openam-shared/src/main/java/com/sun/identity/shared/encode/CookieUtils.java:479]: Could be cookie set without domain?

File: openam-shared/src/main/java/com/sun/identity/shared/encode/CookieUtils.java [hunk 2]
```diff
@@ -474,10 +476,17 @@ public static void addCookieToResponse(HttpServletResponse response, Cookie cook
...
                 domains.add(domain);
             }
         }
+        //IE fix: allow add .app.domain.com after .domain.com without SystemPropertiesManager.getAsBoolean(Constants.SET_COOKIE_TO_ALL_DOMAINS, true)
```
Comment (maximthomas) [inline openam-shared/src/main/java/com/sun/identity/shared/encode/CookieUtils.java:483]: If you want to retreive cookies in particular order, you should use LinkedHashSet instand of HashSet

File: openam-shared/src/main/java/com/sun/identity/shared/encode/CookieUtils.java (hunk 1 above)
Comment (vharseko) [reply openam-shared/src/main/java/com/sun/identity/shared/encode/CookieUtils.java:479]: domains.add(domain); dont allow null

Comment (maximthomas) [reply openam-shared/src/main/java/com/sun/identity/shared/encode/CookieUtils.java:479]: yes, you are right

**Pull request merged**